cache/surrogate/specification.md

```diff
@@ -50,6 +50,29 @@ interpreted as described in [@!RFC2119].
 *  Cache: The system that will store the resources, handle then incoming requests and try to serve the content matching
    a key if it already has in his storage.
 
+This specification defines the following key parameter:
+
+stored       = sf-boolean
+bypass       = sf-boolean
+detail       = sf-token / sf-string
+
+## The stored directive
+"stored", when true, indicates that the key has been stored and can be used and will be treated while passed in a PURGE 
+request.  
+"stored" and "bypass" are exclusive; only one of them should appear on each list member.
+
+## The bypass directive
+"bypass", when true, indicates that the key has not been stored because it was not satisfied by the validation rules.
```

Comment:
Do you have a real-world example for this?

It is not something Varnish or Fastly offers or other HTTP cache solutions? (in relation to Surrogate Keys)

Is there a reason you do not want to treat `Surrogate-Key` header values like popular CDNs Fastly and Cloudflare (`Cache-Tag`) do?

Reply:
Because your storage could be down, the sent `Surrogate-Key` is like `ALL_KEYS` which is a reserved prefix in your implementation  for example.

Reply:
I'm sorry I don't follow. "storage could be down"?

In the context of `Surrogate-Key` header, I believe it should follow the well established usage already out there. Why do you want to do this differently?

Reply:
I'm trying to normalise things and stop being specific provider dependant. Because Varnish is doing in a way, fastly is doing another one and cloudflare too.

Reply:
And because it could be cool to have an interoperable system.

Reply:
Conversation moving to https://github.com/darkweak/souin/pull/134#discussion_r706713492 thread.

Summary, instead of normalizing by adopting an existing implementation, you want to create a new implementation that doesn't first identify issues with the existing ones. Perhaps they all had the same stance as well.

> And because it could be cool to have an interoperable system.

If you agree with that way, then you should be onboard with `CDN-Cache-Control`/`Surrogate-Control` being a `Cache-Control` equivalent override to a service like Souin. That's what the industry leaders have collaborated on, choosing to differ from that is the opposite of interoperability and normalizing.

---

Please provide explanation of "storage could be down". What is the request/response flow where this makes a difference with what you're proposing? How does it help a client such as a web browser?

Reply:
RFC redaction dropped.